src/codecs/encoding/transformer.rs

```diff
@@ -123,25 +125,12 @@ impl Transformer {
 
     fn apply_only_fields(&self, log: &mut LogEvent) {
         if let Some(only_fields) = self.only_fields.as_ref() {
-            let mut to_remove = match log.keys() {
-                Some(keys) => keys
-                    .filter(|field| {
-                        let field_path = parse_value_path(field);
-                        !only_fields
-                            .iter()
-                            .any(|only| field_path.segments.starts_with(&only.segments[..]))
-                    })
-                    .collect::<Vec<_>>(),
-                None => vec![],
-            };
-
-            // reverse sort so that we delete array elements at the end first rather than
-            // the start so that any `nulls` at the end are dropped and empty arrays are
-            // pruned
```

Comment:
I'm assuming this behavior is tested and kept as-is? Or is this no longer relevant?

Reply:
This didn't actually work correctly to begin with, which is one reason I removed it, and it's no longer needed since a different approach was taken. Instead of getting all the fields to remove, it simply creates a new empty event and applies the desired fields.

Previously, this was sorting the string format of paths in reverse order, assuming that would put "deeper" paths first, which is not accurate.